packages/react/react/src/components/Portal/Portal.tsx

```diff
@@ -0,0 +1,80 @@
+import { createContext, useCallback, useContext, useLayoutEffect, useMemo } from 'react';
+import { createPortal } from 'react-dom';
+import { useIsMounted } from '../../hooks/useIsMounted';
+
+interface PortalProps {
+  children: React.ReactNode;
+  className?: string;
+  containerRef?: React.RefObject<HTMLElement | null>;
+}
+
+const portalContext = createContext<{ parentPortal: HTMLElement | null }>({
+  parentPortal: null,
+});
+
+const PORTAL_DEFAULT_CLASS = 'portal';
+
+function RenderPortal({ children, className, containerRef }: PortalProps) {
+  const { parentPortal } = useContext(portalContext);
+
+  const getPortalNode = useCallback(
+    (mountNode: HTMLElement) => {
+      const portalNode = mountNode.ownerDocument.createElement('div');
+      portalNode.classList.add(className || PORTAL_DEFAULT_CLASS);
+
+      return portalNode;
+    },
+    [className]
+  );
+
+  /**
+   * This is the mount node to render portal nodes.
+   * The mountNode has the value "containerRef.current" if it has a "containerRef", or the parent portal node if it is a nested portal.
+   * By default, it has "document.body".
+   */
+  const mountNode = useMemo(() => {
+    if (parentPortal) {
+      return parentPortal;
+    }
+
+    if (containerRef?.current) {
+      return containerRef.current;
+    }
+
+    return document.body;
+  }, [parentPortal, containerRef]);
+
+  const portalNode = useMemo(() => getPortalNode(mountNode), [getPortalNode, mountNode]);
+
+  useLayoutEffect(() => {
+    mountNode.appendChild(portalNode);
+
+    /**
+     * "portalNode" is removed from "mountNode" on unmount.
+     */
+    return () => {
+      if (mountNode.contains(portalNode)) {
+        mountNode.removeChild(portalNode);
+      }
+    };
+  }, [portalNode, mountNode]);
+
+  return createPortal(
+    <portalContext.Provider value={{ parentPortal: portalNode }}>{children}</portalContext.Provider>,
+    portalNode
+  );
+}
+
+/** @tossdocs-ignore */
+export function Portal({ children, ...restProps }: PortalProps) {
+  const isMounted = useIsMounted();
+
+  /**
+   * With this code, it is possible to solve the "window is not defined" and "Hydration Error" that can occur in SSR.
+   */
+  if (!isMounted) {
+    return <></>;
```

Comment:
@ChanhyukPark-Tech 
My intention is simply that Portal's return type is inferred to be 'JSX.Element | null'. 

If you use Fragment, it will be inferred as 'JSX.Element' instead of 'JSX.Element | null'.

But neither is a big deal functionally... 👍

Reply:
neither is a big deal functionally... but Isn't it better to express it clearly to user: `this component can return nothing` by using null?

https://react.dev/learn/conditional-rendering#conditionally-returning-nothing-with-null

I could understand your intention, but I want to know why It's better `JSX.Element` than `JSX.Element | null`.
because I thought it's more clear,`JSX.Element | null`. I want to know case that why you need just `JSX.Element`. (I'm not sure if this is correct, so I'm curious about what many people think. 
@ssi02014, @ChanhyukPark-Tech, @raon0211, @HoseungJang etc.)

Reply:
@manudeli
Thanks for the feedback. 🙏

I too was initially torn between null and Fragment.

My initial intent was simply that the Portal would render based on whether it was mounted or not, so I assumed that it would always render unless there were special circumstances, such as an error.

So, I felt it was awkward to have null as a return type, and since neither null nor Fragment were functionally problematic, I chose Fragment.

However, I'm going to defer to the majority here because I don't think this debate is a functional issue, but a matter of preference. 🙇‍♂️